helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml

```diff
@@ -176,7 +178,13 @@ spec:
       volumes:
         - name: socket-path
           hostPath:
-            path: {{ .Values.chaosDaemon.socketPath | default "/var/run/docker.sock" }}
```

Comment:
If the user's container runtime has a different socket path (e.g. he set it manually), he will not be able to modify the socket path 🤔 , which is provided in the existed configuration. 

Or is there a way to be compatible with the old configuration? For example, automatically get the `dirname` of the path. @cwen0  @STRRL how do you think about it?

Reply:
> Or is there a way to be compatible with the old configuration? 
yes, we can use `dir` func in helm template to be compatible with the old configuration.

and at the same time, we can also provide a arg to let users customize socket dir, called `.Values.chaosDaemon.socketdir` or something.

Reply:
@YangKeao I use `dir` func in helm template to be compatible with the old configuration.

and also provide a arg called `.Values.chaosDaemon.socketDir` to let users customize socket dir.

PTAL